### DISCUSSION TITLE
Codespaces constantly require me to manually delete `.git/index.lock`

### DISCUSSION THREAD

**Discussion by Milo123459** [community/community#42415] [Codespaces]
### Select Topic Area

Bug

### Body

Hello. Codespaces have become increasingly annoying to deal with due to me having to constantly delete `.git/index.lock` almost anytime I want to perform a git operation in the CLI. I'm not sure what is creating the lock / what operation, but I presume it to be something in VSCode.

For more information, I am using starship prompt and perhaps that could be interfering. This is just getting exceptionally annoying because I work on many branches of the same projects at once.

**Comment by erikaellr** [comment]
Hi @Milo123459, we're thrilled you're using Codespaces! Sorry to hear you've been experiencing this annoyance. I wasn't able to reproduce it in a codespace of my own, but `.git/index.lock` will only get created by a Git process, so presumably something in your environment is running a Git process when you don't want it to be, perhaps a VSCode extension or a shell integration. Something you might try is using [VSCode's Extension Bisect utility](https://code.visualstudio.com/blogs/2021/02/16/extension-bisect) to see if the problem is with one of your extensions. Hope this helps, and good luck!
